### PR TITLE
Limit length on PhoneFormField

### PR DESCRIPTION
**Pull request by martinpelli**
This PR adds a new property `limitLength` to `PhoneFormField` so user is not able to input more digits when the selected country doesn't requires more. It is by default in `false`.

## Checklist

Closes [#261](https://github.com/cedvdb/phone_form_field/issues/261)

- [x] I have bumped the version in pubspec.yaml
- [x] I have added an entry in changelog.md for the new pubspec version
- [x] (if applicable) I have added "Closes #1234" to this termplate to automatically close related issues.
- [x] (not required if no sensible change has been made eg: Localization) I have added tests that prove my fix is effective or that my feature works.

### CODE REVIEW

File: lib/src/phone_form_field_state.dart [hunk 1]
```diff
@@ -225,6 +236,59 @@ class PhoneFormFieldState extends FormFieldState<PhoneNumber> {
     }
     return padding;
   }
+
+  void _changeMaxValidLength() {
```
Comment (cedvdb) [inline lib/src/phone_form_field_state.dart:240]: Could you add an explanation regarding formatting ? Because the length could differ depending on the format used right ?

Comment (martinpelli) [reply lib/src/phone_form_field_state.dart:254]: Hi, I added an explanation in my last commit, I had some comments inside the method but I think with a complete one is much clear. Check it out and let me know.

File: test/phone_form_field_test.dart [hunk 2]
```diff
@@ -697,6 +699,22 @@ void main() {
         await tester.pumpAndSettle();
         expect(find.text(national), findsNothing);
       });
+
+      testWidgets(
+          'Should cut off the TextField value when limitLength is true and a long nsn for the selected country is provided',
```
Comment (cedvdb) [inline test/phone_form_field_test.dart:704]: nit:

"Should cut off the TextField value when limitLength is true and a long nsn for the selected country is provided"

To

"Should cut off the TextField value when limitLength is true and an nsn longer than the limit for the selected country is provided"

File: test/phone_form_field_test.dart [hunk 3]
```diff
@@ -697,6 +699,22 @@ void main() {
...
+          (tester) async {
+        PhoneNumber? phoneNumber = PhoneNumber.parse('+32');
```
Comment (cedvdb) [inline test/phone_form_field_test.dart:706]: Adding a test for a phone number format that contains  the character "(" like the US would be helpful to ensure this works.

File: lib/src/phone_form_field_state.dart (hunk 1 above)
Comment (cedvdb) [reply lib/src/phone_form_field_state.dart:254]: Thanks for the detailed explanation. I've a question regarding the logic:

Currently the formatted value is used for the length because the limit is provided in the `maxLength` parameter of `TextField`. Is there a reason to not use a `TextInputFormatter ` instead so we can use the unformatted value inside that formatter rather than the formatted value which is a bit less safe ?

File: test/phone_form_field_test.dart (hunk 3 above)
Comment (martinpelli) [reply test/phone_form_field_test.dart:706]: If you enter an US number it will work, the problem is when you paste it, it won't work because the number is invalid and shorter (because is not formatted). And I can't simulate a test with real typing, because tester.enterText will replace the whole value (like if the user is pasting it).

**Comment by martinpelli** [comment]
@cedvdb I added a new commit with a new implementation with `TextInputFormattter`. Now is much cleaner and it is supporting pasting values, so I was able to add a new test. I do understand your worry, but I think with this new implementation we are removing those edge cases. Let me know if you don't want to merge it as I will need to use my own fork on my app.

**Comment by cedvdb** [comment]
Thanks, it indeed removes the edge cases

**Comment by cedvdb** [comment]
released on pubspec

**Comment by martinpelli** [comment]
@cedvdb I found an issue, the text selection is breaking when the cursor is in the last character and the formatter cuts it. Should I open a new PR and update the changelog file with a new version, or do you prefer to includes it in 10.0.8 or how do you want to handle it?

**Comment by cedvdb** [comment]
A new PR would be best
